contracts/contracts/buyback/Buyback.sol

```diff
@@ -51,25 +67,70 @@ contract Buyback is Governable {
      * @dev Execute a swap of OGN for OUSD via Uniswap or Uniswap compatible
      * protocol (e.g. Sushiswap)
      **/
-    function swap() external onlyVault {
-        if (uniswapAddr == address(0)) return;
-
+    function swap() external onlyVault nonReentrant {
         uint256 sourceAmount = ousd.balanceOf(address(this));
         if (sourceAmount < 1000 * 1e18) return;
+        if (uniswapAddr == address(0)) return;
```

Comment:
Not really related to this PR but might be nice to add: with the setUniswapAddr we should only run the safeApprove functions on uniswapAddr if address is not the zero address.

Reply:
Done